pkg/apis/core/types.go

```diff
@@ -583,9 +583,10 @@ type StorageMedium string
 
 // These are the valid value for StorageMedium
 const (
-	StorageMediumDefault   StorageMedium = ""          // use whatever the default is for the node
-	StorageMediumMemory    StorageMedium = "Memory"    // use memory (tmpfs)
-	StorageMediumHugePages StorageMedium = "HugePages" // use hugepages
+	StorageMediumDefault         StorageMedium = ""           // use whatever the default is for the node
+	StorageMediumMemory          StorageMedium = "Memory"     // use memory (tmpfs)
+	StorageMediumHugePages       StorageMedium = "HugePages"  // use hugepages
+	StorageMediumHugePagesPrefix StorageMedium = "HugePages-" // prefix for full medium notation HugePages-<size>
```

Comment:
since this is adding support for variable hugepage size volumes, we will hit an issue with apiserver and kubelet node skew that we would not have hit prior.  i think this should be gated by a new feature gate as a result.  Can we add a feature gate for "HugePageStorageMediumSize" storage volume that would gate persistence of HugePages-<size>?  We could then default the feature gate to on by default +2 releases from this release.

Reply:
@derekwaynecarr thank you for the review! I'll implement feature gate as you've suggested.

BTW, would it make sense to move hugepage medium validation code from pkg/volume/emptydir/ to pkg/apis/core/validation/  to prevent scheduling pods with invalid hugepage medium (see details [here](https://github.com/kubernetes/kubernetes/pull/84051#discussion_r336060531))?